### PR TITLE
[bug] MatrixType bug fix: Fix numerical error with grouped ndrange

### PR DESCRIPTION
**Pull request by jim19930609**
Issue: https://github.com/taichi-dev/taichi/issues/5819

### CODE REVIEW

File: python/taichi/lang/ast/ast_transformer.py [hunk 1]
```diff
@@ -1173,9 +1173,13 @@ def build_grouped_ndrange_for(ctx, node):
                     f"Group for should have 1 loop target, found {len(targets)}"
                 )
             target = targets[0]
-            target_var = impl.expr_init(
-                matrix.Vector([0] * len(ndrange_var.dimensions),
-                              dt=primitive_types.i32))
+            if current_cfg().real_matrix:
+                mat = matrix.make_matrix([0] * len(ndrange_var.dimensions))
```
Comment (AD1024) [inline python/taichi/lang/ast/ast_transformer.py:1177]: Maybe good to pass a `dt` to make_matrix here, too.

**Review by AD1024 — APPROVED** [review]
LGTM!

File: python/taichi/lang/ast/ast_transformer.py (hunk 1 above)
Comment (jim19930609) [reply python/taichi/lang/ast/ast_transformer.py:1177]: thanks！

**Review by AD1024 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**